x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/services/amazonbedrock/AmazonBedrockChatCompletionRequestManager.java

```diff
@@ -46,9 +51,45 @@ public void execute(
         Supplier<Boolean> hasRequestCompletedFunction,
         ActionListener<InferenceServiceResults> listener
     ) {
-        var chatCompletionInput = inferenceInputs.castTo(ChatCompletionInput.class);
-        var inputs = chatCompletionInput.getInputs();
-        var stream = chatCompletionInput.stream();
+        switch (inferenceInputs) {
```

Comment:
How about we create a new class that handles the `UnifiedChatInput` directly?

Once we've done that we can use that class directly in `doUnifiedCompletionInfer`.